pallets/parachain-staking/src/lib.rs

```diff
@@ -1794,7 +1815,7 @@ pub mod pallet {
 				Error::<T>::NomStakeBelowMin
 			);
 
-			Self::prep_unstake(&delegator, less)?;
+			Self::prep_unstake(&delegator, less, T::MaxUnstakeRequests::get().saturating_sub(1))?;
```

Comment:
here too

Reply:
Based on @Diiaablo95 's discomfort I refactored the limit param of `prep_unstake` in 0e67cbd. I prefer this and hope so do you.